modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/indexer/IndexerPermissionPostFilterImpl.java

```diff
@@ -47,36 +45,31 @@ public IndexerPermissionPostFilterImpl(
 	public boolean hasPermission(
 		PermissionChecker permissionChecker, long entryClassPK) {
 
-		Optional<ModelResourcePermission<?>> optional =
+		ModelResourcePermission<?> modelResourcePermission =
```

Comment:
@jeffwu0724 It seems the supplier can return `null` based on `isPermissionAware()`, but you don't handle it after converting to non `Optional`. This will result in unhandled NPEs. Same case with `isVisible`.